### PR TITLE
Awesome lint

### PR DESCRIPTION
**Pull request by pmalirz**
Enable awesome-lint with github workflows

### CODE REVIEW

**Comment by pmalirz** [comment]
Hi @rrrutledge, @spier 
In order to make the linting green please fix this:
 ✖  1:1  The repository should have "**awesome**" as a GitHub topic       remark-lint:awesome-github
 ✖  1:1  The repository should have "**awesome-list**" as a GitHub topic  remark-lint:awesome-github
It is about adding two tags like here: https://github.com/sindresorhus/awesome ("About" section)
It can be done from the settings level.
Thanks

**Comment by spier** [comment]
@pmalirz done.

Look what GitHub suggested I do In the settings :)

Screenshot 2022-11-17 at 11 40 05

**Comment by rrrutledge** [comment]
Thanks, @spier ❗️

**Comment by spier** [comment]
@pmalirz I re-ran the action. Looks like the check is green now.

There is some warning in the output of the GHA run:
> Node.js 12 actions are deprecated.

I will try and see if I can fix that myself.

**Comment by spier** [comment]
Sorry. I closed this by accident.

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  pull_request:
+    branches: [main]
+jobs:
+  Awesome_Lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - run: npx awesome-lint
```
Comment (spier) [inline .github/workflows/main.yaml]:
```suggestion
  Awesome_Lint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
      - run: npx awesome-lint

```

Using latest version of `checkout`

**Comment by pmalirz** [comment]
> @pmalirz done.
> 
> Look what GitHub suggested I do In the settings :)
> 
> Screenshot 2022-11-17 at 11 40 05

That is nice :)

File: .github/workflows/main.yaml (hunk 1 above)
Comment (pmalirz) [reply .github/workflows/main.yaml]: 👍

**Comment by pmalirz** [comment]
@spier checks are passing so I think it is ok now

**Comment by spier** [comment]
@pmalirz yeah, and I also fixed the warning about node12.

I will do one more fix and then merge this. Great stuff!

**Comment by pmalirz** [comment]
@spier Great

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: CI
+on:
+  pull_request:
+    branches: [main]
+jobs:
+  Awesome_Lint:
```
Comment (spier) [inline .github/workflows/main.yaml]:
```suggestion
  awesome-lint:
```

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+name: CI
```
Comment (spier) [inline .github/workflows/main.yaml:1]: @pmalirz `CI` is a pretty generic name for this GitHub Action.

Do you want to add further Ci jobs to this file?
Or if you want to keep this focused on awesome lint, then we could change it to `name: Awesome Lint` possibly.

But your call, I have only used GitHub Actions a little bit, so I don't know what works best here :)

**Review by spier — APPROVED** [review]

File: .github/workflows/main.yaml (hunk 3 above)
Comment (pmalirz) [reply .github/workflows/main.yaml:1]: @spier this config was copied from the reference / sugested awesome-lint repository... it's been working fin on my local and on the branch so I left it in that shape... of course I'm happy to tune it as per your suggestions, not a problem.

Comment (pmalirz) [reply .github/workflows/main.yaml:1]: Yeah, I didn't study github actions as well

Comment (spier) [reply .github/workflows/main.yaml:1]: Let's keep it as is then, and change it in the future if we see a need.

**Comment by spier** [comment]
@pmalirz all done and merged.

Thanks for adding the linter!

**Comment by pmalirz** [comment]
@spier Thanks for reviewing the PR

**Pull request merged**